web/src/hooks/useTodos.tsx

```diff
@@ -15,8 +14,7 @@ const useTodos = (): {
   const [todos, setTodos] = useState<AddTodoResponse[]>([])
   const [isLoading, setLoading] = useState<boolean>(true)
   const [error, setError] = useState<AxiosError<ErrorResponse> | null>(null)
-  const { token } = useContext(AuthContext)
-  const todoAPI = new TodoAPI(token)
+  const todoAPI = useTodoAPI()
```

Comment:
When the missing dependency, todoAPI, was added to the dependency-array in the useEffect, this initialisation would've caused an infinite loop of re-renders. However, by separating it in a custom hook, it will only re-initialise when the token value is updated. I guess it's also separation of concerns, because it wouldn't make sense that a tiny custom hook component had ownership of the ??whole API??